node/cn/gasprice/gasprice_test.go

```diff
@@ -0,0 +1,232 @@
+	return nil, nil
+}
+func (mock *backendMock) GetBlockReceipts(ctx context.Context, blockHash common.Hash) types.Receipts {
+	return nil
+}
+
+func (mock *backendMock) GetTxLookupInfoAndReceipt(ctx context.Context, hash common.Hash) (*types.Transaction, common.Hash, uint64, uint64, *types.Receipt) {
+	return nil, common.Hash{}, 0, 0, nil
+}
+
+func (mock *backendMock) GetTxAndLookupInfo(hash common.Hash) (*types.Transaction, common.Hash, uint64, uint64) {
+	return nil, common.Hash{}, 0, 0
+}
+
+func (mock *backendMock) GetTd(blockHash common.Hash) *big.Int {
+	return nil
+}
+
+func (mock *backendMock) GetEVM(ctx context.Context, msg blockchain.Message, state *state.StateDB, header *types.Header, vmCfg vm.Config) (*vm.EVM, func() error, error) {
+	return nil, nil, nil
+}
+
+func (mock *backendMock) SubscribeChainEvent(ch chan<- blockchain.ChainEvent) event.Subscription {
+	return nil
+}
+
+func (mock *backendMock) SubscribeChainHeadEvent(ch chan<- blockchain.ChainHeadEvent) event.Subscription {
+	return nil
+}
+
+func (mock *backendMock) SubscribeChainSideEvent(ch chan<- blockchain.ChainSideEvent) event.Subscription {
+	return nil
+}
+
+func (mock *backendMock) IsParallelDBWrite() bool {
+	return false
+}
+
+func (mock *backendMock) GetNonceInCache(address common.Address) (uint64, bool) {
+	return 0, false
+}
+
+func (mock *backendMock) IsSenderTxHashIndexingEnabled() bool {
+	return false
+}
+
+// TxPool API
+func (mock *backendMock) SendTx(ctx context.Context, signedTx *types.Transaction) error {
+	return nil
+}
+
+func (mock *backendMock) GetPoolTransactions() (types.Transactions, error) {
+	return nil, nil
+}
+func (mock *backendMock) GetPoolTransaction(txHash common.Hash) *types.Transaction {
+	return nil
+}
+
+func (mock *backendMock) GetPoolNonce(ctx context.Context, addr common.Address) uint64 {
+	return 0
+}
+
+func (mock *backendMock) Stats() (pending int, queued int) {
+	return 0, 0
+}
+
+func (mock *backendMock) TxPoolContent() (map[common.Address]types.Transactions, map[common.Address]types.Transactions) {
+	return nil, nil
+}
+
+func (mock *backendMock) SubscribeNewTxsEvent(chan<- blockchain.NewTxsEvent) event.Subscription {
+	return nil
+}
+
+func (mock *backendMock) ChainConfig() *params.ChainConfig {
+	return nil
+}
+
+func (mock *backendMock) CurrentBlock() *types.Block {
+	return nil
+}
+
+func (mock *backendMock) GetTxAndLookupInfoInCache(hash common.Hash) (*types.Transaction, common.Hash, uint64, uint64) {
+	return nil, common.Hash{}, 0, 0
+}
+
+func (mock *backendMock) GetBlockReceiptsInCache(blockHash common.Hash) types.Receipts {
+	return nil
+}
+
+func (mock *backendMock) GetTxLookupInfoAndReceiptInCache(Hash common.Hash) (*types.Transaction, common.Hash, uint64, uint64, *types.Receipt) {
+	return nil, common.Hash{}, 0, 0, nil
+}
+
+func TestGasPrice_NewOracle(t *testing.T) {
+	mock := &backendMock{}
+	params := Config{}
+	oracle := NewOracle(mock, params)
+
+	assert.Nil(t, oracle.lastPrice)
```

Comment:
Minor comment:
Duplicated test code can be simplified with test parameters and a for-loop.